### PR TITLE
Fix: JS API Refactor Backward Compatibility [ED-6692]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/common/assets/js/api/modules/command-infra.js [hunk 1]
```diff
@@ -60,6 +60,15 @@ export default class CommandInfra extends ArgsObject {
...
 
+		// TODO - Remove backwards compatibility.
+		Object.defineProperty( this, 'currentCommand', {
+			get() {
+				elementorCommon.helpers.softDeprecated( 'this.currentCommand', '3.7.0', 'this.command' );
+
+				return this.command;
+			},
+		} );
+
```
Comment (StyleShit) [inline core/common/assets/js/api/modules/command-infra.js:71]: Why do you need `defineProperty`? Can't you do it like that?

```JS
get currentCommand() { ... }
```

File: core/common/assets/js/api/core/routes.js [hunk 2]
```diff
@@ -105,6 +105,8 @@ export default class Routes extends Commands {
 			args.onBefore.apply( component, [ args ] );
 		}
 
+		this.trigger( 'run:before', component, route, args );
```
Comment (StyleShit) [inline core/common/assets/js/api/core/routes.js:108]: Shouln't we deprecate it?

File: core/common/assets/js/api/core/routes.js [hunk 3]
```diff
@@ -144,14 +146,16 @@ export default class Routes extends Commands {
...
+	afterRun( route, args, results = undefined ) {
 		const component = this.getComponent( route );
 
 		component.onRoute( route, args );
 
 		if ( args.onAfter ) {
 			args.onAfter.apply( component, [ args ] );
 		}
+
+		this.trigger( 'run:after', component, route, args, results );
```
Comment (StyleShit) [inline core/common/assets/js/api/core/routes.js:158]: Same as above

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/routes.js (hunk 2 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/routes.js:108]: We don't have hooks on `$e.routes` yet.

**Review by StyleShit — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**